code/modules/clothing/under/_under.dm

```diff
@@ -23,6 +23,16 @@
 	var/mutantrace_variation = NO_MUTANTRACE_VARIATION //Are there special sprites for specific situations? Don't use this unless you need to.
 	var/freshly_laundered = FALSE
 
+/obj/item/clothing/under/Initialize(mapload)
+	. = ..()
+	if(random_sensor)
+		//make the sensor mode favor higher levels, except coords.
+		sensor_mode = pick(SENSOR_OFF, SENSOR_LIVING, SENSOR_LIVING, SENSOR_VITALS, SENSOR_VITALS, SENSOR_VITALS, SENSOR_COORDS, SENSOR_COORDS)
+
+/obj/item/clothing/under/Destroy()
+	dump_attachment()
+	return ..()
```

Comment:
`Destroy()` should really leave behind nothing when it's done, and not have side effects like ejecting items. 
Accessory / attachment removal should be handled in `deconstruct()` for clothes, for example. 

An alternative should be found involving the holodeck's `derez` proc, and how it removes and deletes spawns atoms - cc @Kylerace

Reply:
`Destroy()` has quite a few cases where it ejects items when the src object is a container.

https://github.com/tgstation/tgstation/blob/6a8ad0ccd38b569d9f8d1158a1debdd374b33401/code/game/objects/items/storage/storage.dm#L54-L59

I'm aware of the `derez` proc and looked into it before determining `Destroy` was the better solution.  What if you take a jumpsuit and light it on fire?  Or stick it into the recycler?  Or spray it with acid?  The holodeck despawning was one of many ways to destroy the object and we should account for everything by putting the eject code in that proc.

I'm not sure why it's harddeleting.  Maybe it's because of this?  

https://github.com/tgstation/tgstation/blob/77c9485ecb4df0d41126d4d6836cc185801df166/code/modules/jobs/job_types/captain.dm#L53-L77

The `dump_attatchment` proc should be cleaning up our vars properly... unless I'm missing something here?

https://github.com/tgstation/tgstation/blob/8a769056d41c8fd08281694377925924d300aefe/code/modules/clothing/under/_under.dm#L337-L349

Reply:
> Destroy() has quite a few cases where it ejects items when the src object is a container.

This is bad and we're trying to remove it, we want `Destroy()` to do one thing: Remove the item and any traces of it from existence

Reply:
Applied with https://github.com/tgstation/tgstation/pull/65768/commits/32121b4127c032940f0a39d4a138a5a81d47e1a7

I still think it's not optimal since you could potential have one big IF() loop for every edge case object that uses it's own custom storage code like cards, linen bins, paper bins, boots, etc.

Reply:
see #63841 for more info on the subject